core-to-plc/src/Language/Plutus/Lift.hs

```diff
@@ -0,0 +1,254 @@
+-- needed to allow the definition of 'typeRep'
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DefaultSignatures   #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
+module Language.Plutus.Lift (LiftPlc (..)) where
+
+import           PlutusPrelude                      (strToBs)
+
+import           Language.Plutus.CoreToPLC.Builtins
+
+import           Language.PlutusCore
+import           Language.PlutusCore.MkPlc
+
+import           Control.Monad
+
+import qualified Data.ByteString.Lazy               as BSL
+import           Data.List                          (elemIndex, sortBy)
+import           Data.Maybe                         (fromJust)
+import           GHC.Generics
+
+-- | Class for types which can be lifted into Plutus Core.
+class LiftPlc a where
+    -- | Get the Plutus Core type corresponding to this type.
+    typeRep :: (MonadQuote m) => m (Type TyName ())
+    default typeRep :: (MonadQuote m, GLiftPlc (Rep a)) => m (Type TyName ())
+    typeRep = gTypeRep @(Rep a)
+
+    -- | Get a Plutus Core term corresponding to the given value.
+    lift :: (MonadQuote m) => a -> m (Term TyName Name ())
```

Comment:
This is so overloaded. Since the class is called `LiftPlc`, maybe call this `liftPlc`?

Reply:
Eh, I like the analogy with `Lift` and `Typeable`. I don't like adding pointless name suffixes when things can easily be disambiguated with qualified imports.